conans/test/unittests/client/build/cmake_test.py

```diff
@@ -188,6 +188,32 @@ def cmake_generator_test(self):
             cmake = CMake(conan_file)
             self.assertIn('-G "My CMake Generator"', cmake.command_line)
 
+    def cmake_generator_platform_test(self):
+        conan_file = ConanFileMock()
+        conan_file.settings = Settings()
+
+        with tools.environment_append({"CONAN_CMAKE_GENERATOR": "My CMake Generator",
+                                       "CONAN_CMAKE_GENERATOR_PLATFORM": "My CMake Platform"}):
+            cmake = CMake(conan_file)
+            self.assertIn('-G "My CMake Generator" -A "My CMake Platform', cmake.command_line)
```

Comment:
Create more tests for the combination of values: What happens if I set a generator like Visual Studio 15 2017 and set the CONAN_CMAKE_GNERATOR_PLATFORM too?

Reply:
added more tests as requested